pkg/lwip/contrib/sock/lwip_sock.c

```diff
@@ -640,13 +640,7 @@ ssize_t lwip_sock_sendv(struct netconn *conn, const sock_tx_snip_t *snips,
     }
 #if LWIP_TCP
     else if (tmp->type & NETCONN_TCP) {
-        while (snips) {
-            err = netconn_write_partly(tmp, snips->data, snips->len, 0, (size_t *)(&res));
-            if (err != ERR_OK) {
-                break;
-            }
-            snips = snips->next;
-        }
+        err = netconn_write_partly(tmp, buf->p->payload, buf->p->len, 0, (size_t *)(&res));
```

Comment:
The pbuf inside might be linked as well, so this will only send the contents of the first buffer. Hmm

Reply:
It was however just allocated [a few lines above](https://github.com/RIOT-OS/RIOT/blob/5cb7ee8338d9e678107ecebc4d13798b4dc5493e/pkg/lwip/contrib/sock/lwip_sock.c#L601)

Reply:
There is a `netconn_write_vectors_partly` which takes an array of `struct netvector` (containing ptr/len). That is probably the best (as loop of write_partly might send some data out of order)

Reply:
netbuf internal pbuf is allocated with `PBUF_RAM` so it is one contiguous buffer. So the code as it is now should be fine.

Reply:
Hm `netconn_write_vectors_partly()` would have mapped fine to v1 of this PR, but now I don't have an array of snips anymore.

 The data was already copied into a continuous buffer `buf` for the UDP case, so can't we just use that? It would be unfortunate though if we had to copy that again with `NETCONN_COPY`.

Reply:
Yes, sending the `buf` contents in one go like this is fine.

`NETCONN_COPY` was if we did like in v1 (and not used the buf but only forwarded the pointers)